src/services/WarehouseCrawlDataServices.ts

```diff
@@ -0,0 +1,159 @@
+import puppeteer from 'puppeteer';
+import winston from '../config/winston';
+import request_promise from 'request-promise';
+import cheerio from 'cheerio';
+import { detailPageWarehouseConfig, crawlDetailPageWareTokyoConfig } from '../config/WarehouseCrawlDataConfig';
+
+async function WarehouseCrawlDataServices() {
+  try {
+    const browser = await puppeteer.launch({ headless: false });
+    const page = await browser.newPage();
+    await page.setRequestInterception(true);
+    await page.on('request', (req) => {
+      const typeImageVideo = ['image', 'media'];
+      const videoExtension = ['.mp4', '.avi', '.flv', '.mov', '.wmv'];
+      const url = req.url().toLowerCase();
+      const resourceType = req.resourceType();
+      const splitStringVideo = url.split('.');
+      // total '.' appear in a string
+      const total = splitStringVideo.length - 1;
+      // show last string
+      const videoTail = splitStringVideo[total];
+      if (
+        videoExtension.includes(videoTail) === true ||
+        typeImageVideo.includes(resourceType) === true
+      ) {
+        req.abort();
+      } else {
+        req.continue();
+      }
+    });
+    winston.info('Redirecting to page.on(\'request\')');
+
+    await page.goto('https://www.cbre-propertysearch.jp/industrial/', {
+      waitUntil: 'load',
+      timeout: 30000,
+    });
+    winston.info('Redirecting to [\'https://www.cbre-propertysearch.jp/industrial/\']');
```

Comment:
đang bị lặp cái `https://www.cbre-propertysearch.jp/industrial/`

mà cái log info này nên đặt ở phía trên e nhé

```suggestion
    winston.info('Redirecting to [\'https://www.cbre-propertysearch.jp/industrial/\']');
    await page.goto('https://www.cbre-propertysearch.jp/industrial/', {
      waitUntil: 'load',
      timeout: 30000,
    });
```

Reply:
đã xử lý